flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/socket/SocketWindowWordCount.java

```diff
@@ -65,33 +70,47 @@ public static void main(String[] args) throws Exception {
 		DataStream<String> text = env.socketTextStream(hostname, port, "\n");
 
 		// parse the data, group it, window it, and aggregate the counts
-		DataStream<WordWithCount> windowCounts = text
-
-				.flatMap(new FlatMapFunction<String, WordWithCount>() {
+		DataStream<WordWithCount> windowCounts = text.flatMap(serializableProxy(
```

Comment:
I don't think the example should be changed. Using proxies in operators is a pretty advanced use case, while the example should focus on basic usage.

Reply:
Good comment, original case should be retained. I will add another case.